src/audio/eq_fir/eq_fir.c

```diff
@@ -851,7 +817,7 @@ static const struct comp_driver comp_eq_fir = {
 	.ops = {
 		.create = eq_fir_new,
 		.free = eq_fir_free,
-		.params = eq_fir_params,
+		.params = NULL,
```

Comment:
This is not needed, as compiler will do it for us. Btw, we should also have eq_iir.c ?

Reply:
I think it's there to explicitly show that the default is to be taken instead of being forgotten.

Does eq_iir.c only do the check for comp_verify_params and **nothing** else?

Reply:
@lgirdwood @paulstelian97 I will remove this explicit NULL initialization, actually I wanted as many common things as possible to "be forgotten" (esp. by the effects components developers) if not required to be customized.

eq_iir does this single call only but uses extra flags as I mentioned earlier.

Reply:
@mmaka1 Thanks for correction with IIR. I forgot there's format conversion capability that FIR does not have.

Reply:
What about mixer ? it does not change rate/format ?

Reply:
@lgirdwood mixer and mux use BUFF_PARAMS_CHANNELS.